### ISSUE TITLE
Parakeet streaming missing tokenizer.model in downloaded model directory

### ISSUE THREAD

**Issue by matt-shearing** [peteonrails/voxtype#442]
## Summary

VoxType 0.7.2+ added Parakeet streaming, but `voxtype setup --download --model parakeet-tdt-0.6b-v3` does not fetch the `tokenizer.model` file that the streaming code requires. As a result, enabling `[parakeet] streaming = true` fails immediately at daemon startup with:

```
Error: Transcription error: Whisper initialization failed: Parakeet streaming (ParakeetUnified) init failed: Tokenizer error: Failed to open tokenizer.model: No such file or directory (os error 2)
Streaming requires a TDT v3 model directory containing tokenizer.model.
If you're using a TDT v2 directory, switch to TDT v3 or set
[parakeet] streaming = false to use the batch pipeline.
```

`voxtype setup --download --model parakeet-tdt-0.6b-v3` itself reports `✓ Model ready: parakeet-tdt-0.6b-v3 (2432 MB)` despite the missing file, so users won't notice until they actually try streaming.

## Reproduction

```bash
voxtype setup --download --model parakeet-tdt-0.6b-v3
ls ~/.local/share/voxtype/models/parakeet-tdt-0.6b-v3/
# config.json  decoder_joint-model.onnx  encoder-model.onnx
# encoder-model.onnx.data  vocab.txt
# (tokenizer.model is absent)

# Then enable streaming in config:
# [parakeet]
# streaming = true
systemctl --user restart voxtype  # daemon crashes
```

## Root cause

The downloader is fetching from `istupakov/parakeet-tdt-0.6b-v3-onnx` on HuggingFace. That repo's file list (via the HF API) is:
```
config.json
decoder_joint-model.int8.onnx / decoder_joint-model.onnx
encoder-model.int8.onnx / encoder-model.onnx / encoder-model.onnx.data
nemo128.onnx
vocab.txt
```

It does not include `tokenizer.model`. The upstream NVIDIA `.nemo` archive (`nvidia/parakeet-tdt-0.6b-v3`) does contain one (`902810505b78491d95378fa3000d1bc6_tokenizer.model`, 353 KB), but extracting and dropping it in causes a secondary failure:

```
ERROR Non-zero status code returned while running Gather node. Name:'/decoder/embed/Gather'
Status Message: indices element out of data bounds, idx=8194 must be within the inclusive range [-8193,8192]
```

The NeMo tokenizer's vocab (8194+ entries) doesn't match the istupakov ONNX decoder's embedding (8193 entries). So even if a user manually grabs the NeMo tokenizer, streaming still doesn't work — the SentencePiece vocab and the ONNX decoder vocab are out of sync.

## Suggested fixes

1. Document which Parakeet ONNX repo VoxType expects, and whether `tokenizer.model` is required for streaming
2. Make the downloader fetch a compatible `tokenizer.model` (either by host upstream of the matching tokenizer, or by picking an ONNX repo that already ships one)
3. Validate the model directory at config-load time and emit a clearer error pointing at this issue, rather than failing at first transcription with the cryptic Gather-node error
4. The docs at https://voxtype.io/docs/PARAKEET list the expected files as `encoder-model.onnx, encoder-model.onnx.data, decoder_joint-model.onnx, vocab.txt, config.json` — no mention of `tokenizer.model`, even though the streaming code requires it. Doc update needed.

## Environment

- VoxType 0.7.3
- Backend: `voxtype-onnx-migraphx` (AMD Radeon RX 7900 XT)
- Arch Linux

Reported via voxtype-tray.

**Comment by matt-shearing** [comment]
## Update — root cause is not the tokenizer, it's a streaming-code bug

After more investigation, the missing `tokenizer.model` is a real packaging gap, but it's not what's actually breaking streaming. Here's the architecture I traced via `onnx.load()` on the istupakov decoder:

```
decoder_joint-model.onnx
  decoder.prediction.embed.weight: dims=[8193, 640]   <-- accepts indices 0-8192 only
  output 'outputs':                shape=[..., 8198]  <-- 8192 subwords + 1 blank + 5 duration
```

So the joint network emits **8198 logits** = subwords[0..8191] + blank[8192] + **duration_tokens[8193..8197]**. This is standard for a TDT (Token-and-Duration Transducer) model: the joint head predicts both *which subword* AND *how many encoder frames to skip*, then the duration tokens are consumed by the streaming loop — they're NOT fed back into the prediction network's embedding lookup.

The streaming code is feeding `idx=8194` (= duration token #2) into `/decoder/embed/Gather`, which only has 8193 rows. That's the bug:

```
ERROR ... Gather node ... idx=8194 must be within the inclusive range [-8193,8192]
```

### What this means

- **A custom tokenizer.model cannot fix this** — the SP tokenizer max-encodes to 8191 (verified by loading the proto and encoding test strings; vocab_size=8192, byte_fallback=False, no BOS/EOS configured). The bad ID is generated inside VoxType's streaming inference loop, not by the tokenizer.
- **Switching to another ONNX repo won't fix it either** — every TDT v3 conversion has the same architecture; the bug is in how VoxType post-processes the joint-network output during streaming.
- **The batch pipeline works correctly** — it correctly separates subword/blank logits from duration logits when computing the next decoder input.

### Suggested fix

In the `ParakeetUnified::transcribe_chunk` streaming code path, when looking up the next decoder input embedding, mask out duration tokens before argmax — i.e., only argmax over `outputs[..., :8193]`, never over the full `outputs[..., :8198]`. The duration prediction should be taken separately from `outputs[..., 8193:8198]` and used to control encoder frame consumption.

If pseudocode helps:
```rust
// Wrong (current streaming code, presumably):
let next_token = outputs.argmax();          // can return 8193..8197

// Right:
let next_token = outputs[..8193].argmax();  // capped to subwords+blank
let duration   = outputs[8193..8198].argmax();
```

I'm running 0.7.3 with the istupakov ONNX model, so the actual symbol names in your code may differ — but the shape mismatch (Gather on 8193 rows vs output of 8198 logits) localizes the bug to wherever the decoder-input feedback path is constructed.

Closing the loop: the tokenizer.model is still worth bundling for documentation/sanity, but it's a separate concern from the streaming crash.

**Comment by peteonrails** [comment]
Thanks for the deep dive @matt-shearing. v0.7.5 closes this bug.

I just tested streaming on v0.7.5 with the new `parakeet-unified-en-0.6b` model (bobNight repo, added in v0.7.4 via #423 specifically for the streaming use case). Same hardware shape as yours (AMD GPU, MIGraphX EP):

```
INFO Loading Parakeet streaming (ParakeetUnified) model from "/home/pete/.local/share/voxtype/models/parakeet-unified-en-0.6b"
INFO Configuring MIGraphX execution provider for AMD GPU acceleration
INFO Parakeet streaming model loaded in 4.27s (chunk=0.32s, left=5.60s, right=0.32s)
INFO Streaming session started (SIGUSR1)
DEBUG Running: wtype -d 5 -- " Okay"
DEBUG Text output via wtype
...
INFO SIGUSR2 stop while streaming; closing capture and disowning session
```

The bobNight repo's decoder graph has a different topology than the istupakov one: it ships `tokenizer.model` AND its decoder embed is sized to handle the streaming inference loop's index space. Your conclusion ("switching to another ONNX repo won't fix it either") was right for the istupakov family, but bobNight's `parakeet-unified-en-0.6b` is a separate purpose-built model that does fix it.

## What v0.7.5 ships

1. **`parakeet-unified-en-0.6b` in the model registry as the streaming default** (`voxtype setup model` → option 15)
2. **TUI auto-switch**: when a user toggles `[parakeet] streaming = true` in the configure TUI while a non-streaming-compatible model (like `parakeet-tdt-0.6b-v3`) is selected, the Advanced section now auto-switches to `parakeet-unified-en-0.6b` and prompts "Press 'd' to download in the background." Mirrors what would otherwise be a cryptic Gather-error at daemon restart.
3. **Validator fix** so the bobNight naming convention (`encoder.onnx` vs istupakov's `encoder-model.onnx`) passes post-download checks.

## What v0.7.5 does NOT yet fix

If you bypass the auto-switch and manually keep `model = "parakeet-tdt-0.6b-v3"` + `streaming = true` in config.toml, the daemon still hits the duration-token-into-embed error you traced. 

That's a runtime trap; v0.7.6 will add a model-capability check at daemon startup that errors out with a clear message before reaching ONNX Runtime ("model 'parakeet-tdt-0.6b-v3' does not support streaming; set streaming = false or switch to parakeet-unified-en-0.6b").

## Workaround for v0.7.5

```bash
# Switch to the streaming-compatible model
voxtype setup model           # pick option 15: parakeet-unified-en-0.6b
# Or via config.toml:
[parakeet]
model = "parakeet-unified-en-0.6b"
streaming = true
```

Thanks for filing the issue and for the analysis. I'm marking the streaming-on-wrong-model error-message improvement for v0.7.6.
